infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -909,80 +789,64 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
-
-        // Store the pool submission struct
-        poolSubmissions[_challengeId][_poolAddress] = PoolSubmission({
-            submitted: true,
-            claimed: false,
-            stakedKeyCount: totalStakedKeys,
-            winningKeyCount: winningKeyCount,
-            assertionStateRootOrConfirmData: _confirmData
-        });
-
-        // Emit the New Pool Submission event
-        emit NewPoolSubmission(_challengeId, _poolAddress, totalStakedKeys, winningKeyCount);
-	}
-
-    /**
-    * @notice Update Previously Submitted Pool assertion to a challenge.
-    * @dev This function is called internally from the submitMultipleAssertions function or submitPoolAssertion function.
-    * @param _poolAddress The address of the pool.
+    * @notice Update Previously Submitted Bulk assertion to a challenge.
+    * @dev This function is called internally when a user stakes or unstakes keys.
+    * @param _bulkAddress The address of the pool or owner wallet.
     * @param _challengeId The ID of the challenge.
     *
      */
-	function _updatePoolAssertion(
-		address _poolAddress,
+	function _updateBulkAssertion(
+		address _bulkAddress,
 		uint256 _challengeId
 	) internal {
         
-        // Add the number of keys staked in the pool to the total owner staked keys
-        uint256 totalStakedKeys = assignedKeysToPoolCount[_poolAddress];
+        // Initially set the number of keys to the number of keys assigned to the pool (will be 0 if pool does not exist)
+        uint256 numberOfKeys = assignedKeysToPoolCount[_bulkAddress] + assignedKeysOfUserCount[_bulkAddress];
 
-        // Get the stakedAmount of _poolAddress for determining boostFactor
-		uint256 stakedAmount = getMaxStakedAmount(_poolAddress, address(0));
+        // Check if the bulk address is a pool
+        bool isPool = PoolFactory(poolFactoryAddress).poolsCreatedViaFactory(_bulkAddress);
+
+        // If bulk address is not a pool, this means it is a wallet owner with unstaked keys
+        if(!isPool){
+            // Determine the number of keys to be submitted for by the owner
+            // This is calculated by taking the total number of keys owned and subtracting the number of keys staked in pools
+            numberOfKeys = NodeLicense(nodeLicenseAddress).balanceOf(_bulkAddress) - assignedKeysOfUserCount[_bulkAddress];
+        }else {
+            // If the bulk address is a pool, check if the caller is the pool owner or an approved operator
+            require(PoolFactory2(poolFactoryAddress).validateSubmitPoolAssertion(_bulkAddress, msg.sender), "17");
+        }
+
+        require(numberOfKeys > 0, "57");
+
+        // Set the boost factor intially to 100
+        uint256 boostFactor = 100;
+
+        // If the bulk address is a pool, determine the boost factor
+        if(isPool){
+            // Get the stakedAmount of _poolAddress for determining boostFactor
+            uint256 stakedAmount = stakedAmounts[_bulkAddress];
```

Comment:
This needs to get the actual maxStakeAmount using `getMaxStakedAmount`

Reply:
Done!